src/js/utils/pagination.js

```diff
@@ -0,0 +1,53 @@
+import { useCallback, useEffect, useState } from 'react';
+
+export const normalizeShow = (showProp, context) => {
+  let page;
+  let showItem;
+
+  if (['dataTable', 'list'].includes(context)) {
+    // by default, show refers to the index of an item,
+    // but if using pagination, show can take the form of { page: # },
+    // where page refers to the page # to show
+    if (typeof showProp === 'number') showItem = showProp;
+    else if (typeof showProp === 'object' && 'page' in showProp)
+      page = showProp.page;
+  }
+
+  return [page, showItem];
```

Comment:
I'm wondering if it would make things cleaner through the calls to return `{ page, item }`. This would allow us to passing this around as a `show` variable/property.

Reply:
I think we can actually simplify this so it handles the normalizing internally and just returns the expected active page. I've udpated.